### ISSUE TITLE
hi are you still working on this? I'd love to work together on this project...

### ISSUE THREAD

**Issue by Finite-Code** [herocoder14/fireboltt097#1]

**Comment by herocoder14** [comment]
Sure, Contact Me wehero741@gmail.com

**Comment by Finite-Code** [comment]
Hi, just emailed you! sorry for being late, also I was planning to port DOOM to this :)

**Comment by herocoder14** [comment]
Good luck bro, It's closed source and encrypted (that's what redditors told
me).
If you need any help I am ready to work together.

**Comment by herocoder14** [comment]
You can update and repair the watch's firmware through a tool that means
there must be a way to access it's system files.

**Comment by Finite-Code** [comment]
Yes surely! there has to be (a way to access internal files), I've tried
decrypting the binary files but got no luck, given it's a mtk chipset, mtk
phones offer something called the boot ROM mode or simply the BROM mode, in
theory this shall feature something similar too!

And looking at the RES (watch face) flashing instructions of the chinese
PDF (I'll link the PDF for that and the Datasheet too if needed) you put
the watch into mass storage mode and then restart it, meaning all the file
transfers must happen while the watch is ON and some sort of handshake
takes place, or either there is some hardware abstraction layer to hide the
internal directories...

I think the second possibility is the case as the low level code has a line
to hide all directories after the read/write process is complete. We must
exploit it at this stage to gain access to the internals...

And regarding DOOM, once we break into the system it's gonna be a hella lot
easier, one thing that's limiting is the PSRAM it only features 256kb or so
of it, whereas RAM in theory is about 256mb (a lot more than enough).

Trust me it's gonna be lit. 😄

Anyways, I love & hate emails they're good to structure your thoughts but
are a slower form of communication, can we chat somewhere else?

**Comment by Circle-Developer** [comment]
@herocoder14 can you specify how you obtained the firmware? I'm trying to reverse some boat watches and tried way hard to obtain it from their apk but failed..... also if you flashed it how did you do it?
